flask_gae_gcs.py

```diff
@@ -218,28 +216,45 @@ def save_files(fields, validators=None, retry_params=None, bucket_name=None):
             else:
                 result.successful = False
             results.append(result)
-        i += 1
     return results
 
 
 def _upload_fields():
-    '''
-      :returns: List of tuples with the filename & `cgi.FieldStorage` as value.
+    '''Gets a list of files from the request.
+    Uses Flask's request.files to get all files, unless the Content-Type is
+    `text/csv` or `text/plain`: then returns the request body as a FileStorage
+    object, attempting to use Content-Disposition to get a file name.
+
+      :returns: List of tuples of:
+                (field_name, `werkzeug.datastructures.FileStorage`)
     '''
     result = []
-    for key, value in request.files.iteritems():
-        if not isinstance(value, unicode):
-            result.append((key, value))
+
+    content_type = request.headers.get('content-type')
+    mime_type, _ = parse_header(content_type)
+
+    if mime_type in ('text/plain', 'text/csv'):
+        _, params = parse_header(request.headers.get('content-disposition'))
```

Comment:
The docstring suggests that content-disposition is optional, but I assume this'll fail if it's not there.